torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -343,16 +343,23 @@ void ScriptModuleSerializer::serialize(
   // so loading the code does not depend on loading the data
   std::vector<IValue> ivalue_constants(
       constant_table_.begin(), constant_table_.end());
-  writeArchive(
-      c10::ivalue::Tuple::create(ivalue_constants),
-      /*archive_name=*/"constants",
-      /*archive_dir=*/"",
-      /*tensor_dir=*/"constants/");
   if (bytecode_format) {
+    writeArchive(
+        c10::ivalue::Tuple::create(ivalue_constants),
+        /*archive_name=*/"constants",
+        /*archive_dir=*/"",
+        /*tensor_dir=*/"constants/",
+        /*tensor_cdata_naming_scheme=*/true);
```

Comment:
Just a note, this change will introduce jitter between saves & loads of the same model since now the tensor names are no longer always the same 0/1/2/3/4 and instead are based off of the tensor's storage's c-memory pointer

Reply:
@cccclai is there a reason why you didn't also set the `tensor_cdata_naming_scheme` to true for the writeArchive call on `:334`? I might be misunderstanding the flow of mobile serialization, but I know that tensors can be written out to a `data/` folder instead of the `constants/` folder and you'd want those also to use the `tensor_cdata_naming_scheme` and `constants/` location for those tensors also

Reply:
Ah this is a format thing. We actually write tensors to two locations today: the data folder, which contains tensors that are explicitly part of the module state (like parameters and buffers), and the constants folder, which contains tensors that are `prim::Constants` in the graph. 

As far as I understand the major goal of this work is to share the constants between the bytecode and TS code, since the data folder is already shared. Is this right @cccclai?

Reply:
Good question! At this moment, the majority (probably all, I'm just being conservative) of duplicated tensors are between `constants` and `bytecode`. Also, for mobile, there is an `optimize_for_mobile()` step to convert tensors from parameters (in `data`) to `constants`, and ideally, there shouldn't be any duplicated tensors between `data` and `bytecode` after `optimized_for_mobile()` step. Given that it's unlikely to have duplication, I think we can wait until the unify format

Reply:
@suo yeah this statement is correct.
> the major goal of this work is to share the constants between the bytecode and TS code, since the data folder is already shared

Reply:
@suo Yes, we shared data (parameters) but not constants between jit and bytecode. For this PR we also share constants.

Reply:
Awesome, this makes sense then! :) 🚢 🚢